openfecli/commands/gather.py

```diff
@@ -147,42 +147,56 @@ def legacy_get_type(res_fn:os.PathLike|str)->Literal['vacuum','solvent','complex
     else:
         return 'complex'
 
+def _generate_bad_legs_error_message(bad_legs:list[tuple[set[str], tuple[str]]])->str:
+    """Attempt to determine whether the legs belong to a RHFE or RBFE calculation
+    and generate an error message accordingly.
 
-def _generate_bad_legs_error_message(set_vals:set, ligpair)->str:
-    expected_rbfe = {'complex', 'solvent'}
-    expected_rhfe = {'solvent', 'vacuum'}
-    maybe_rhfe = bool(set_vals & expected_rhfe)
-    maybe_rbfe = bool(set_vals & expected_rbfe)
-    if maybe_rhfe and not maybe_rbfe:
-        msg = (
-                "This appears to be an RHFE calculation, but we're "
-                f"missing {expected_rhfe - set_vals} runs for the "
-                f"edge with ligands {ligpair}."
+    Parameters
+    ----------
+    bad_legs : list[set[str], tuple[str]]]
+        A list of tuples of (leg_types, ligpair) pairs from failed edges/legs.
+
+    Returns
+    -------
+    str
+        An error message containing information on all failed legs.
+    """
+    msg="\nThe following legs are invalid:\n"
+    for leg_types, ligpair in bad_legs:
+        expected_rbfe_types = {'complex', 'solvent'}
+        expected_rhfe_types = {'solvent', 'vacuum'}
+        maybe_rhfe = bool(leg_types & expected_rhfe_types)
+        maybe_rbfe = bool(leg_types & expected_rbfe_types)
+        if maybe_rhfe and not maybe_rbfe:
+            msg += (
+                f"{ligpair}:\n"
+                "\tAssuming this is an RHFE calculation, "
```

Comment:
This looks like it might get really noisy, with lots of `Assuming X this is missing`. Could we just set that as a single header and then just `+=` on a list of edge information?

Reply:
@jthorton @IAlibay how do you feel about this solution?

Here's what the output looks like:

![image](https://github.com/user-attachments/assets/53850ec9-8385-4c0a-9050-bef6185059bf)

My thinking is that we pass through as much info to the user as we can without trying to be overly clever and auto-detecting rbfe vs  rhfe, since it sounds like verbosity hurts more than helps. 

Happy to change it again in the future depending on user needs.

Reply:
This looks good, it is clear from this message which edges have an issue and it is readable. What about adding something at the end with info on restarting the calculations if they have failed? 

For the guessing of the calculation type if we need that info we could require the `AlchemicalNetwork` which would explicitly tell us, this might be useful in future if we wanted to also report missing legs from the network not just failed edges.